src/lib/y2firewall/clients/proposal.rb

```diff
@@ -106,15 +108,54 @@ def write
       # Obtain and call the corresponding method for the clicked link.
       def call_proposal_action_for(link)
         action = link.gsub("firewall--", "")
-        @settings.public_send("#{action}!")
+        if action == "cpu_mitigations"
+          bootloader_dialog
+        else
+          @settings.public_send("#{action}!")
+        end
       end
 
       # Array with the available proposal descriptions.
       #
       # @return [Array<String>] services and ports descriptions
       def proposals
-        # Filter proposals with content and sort them
-        [firewall_proposal, sshd_proposal, ssh_port_proposal, vnc_fw_proposal].compact
+        # Filter proposals with content
+        [cpu_mitigations_proposal, firewall_proposal, sshd_proposal,
+         ssh_port_proposal, vnc_fw_proposal].compact
+      end
+
+      # Returns the cpu mitigation part of the bootloader proposal description
+      # Returns nil if this part should be skipped
+      # @return [String] proposal html text
+      def cpu_mitigations_proposal
+        require "bootloader/bootloader_factory"
+        bl = ::Bootloader::BootloaderFactory.current
+        return nil if bl.name == "none"
+
+        mitigations = bl.cpu_mitigations
+
+        res = _("CPU Mitigations: ") + format("<a href=\"%s\">", LINK_CPU_MITIGATIONS) +
+          mitigations.to_human_string + "</a>"
+        log.info "mitigations output #{res.inspect}"
+        res
+      end
+
+      def bootloader_dialog
+        require "bootloader/config_dialog"
+        Yast.import "Bootloader"
```

Comment:
NP: As there is no dependency on `yast2-bootloader` (as this is only for the installer where we can assume it's always present) in might theoretically raise an exception in the `require` here. In that case the `ensure Yast::Wizard.CloseDialog` call would be wrong, the dialog would not be created yet.

I'd add a `begin` statement below.

Reply:
ok, make sense.